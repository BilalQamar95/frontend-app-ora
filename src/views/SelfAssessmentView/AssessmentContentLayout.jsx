import React from 'react';
import PropTypes from 'prop-types';

import { Col, Row } from '@edx/paragon';

import Rubric from 'components/Rubric';
import AssessmentContent from './AssessmentContent';

import './AssessmentContentLayout.scss';

const AssessmentContentLayout = ({
  submission,
  oraConfigData,
}) => (
  <div className="assessment-content-layout mr-auto ml-auto">
    <div className="content-wrapper">
      <Row className="flex-nowrap m-0">
        <Col className="p-0">
          <AssessmentContent
            submission={submission}
            oraConfigData={oraConfigData}
          />
        </Col>
        <Rubric isGrading />
      </Row>
    </div>
  </div>
);

AssessmentContentLayout.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  submission: PropTypes.any.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  oraConfigData: PropTypes.any.isRequired,
};

export default AssessmentContentLayout;
