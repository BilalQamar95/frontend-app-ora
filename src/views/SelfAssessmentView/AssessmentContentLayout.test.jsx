import { shallow } from '@edx/react-unit-test-utils';
import AssessmentContentLayout from './AssessmentContentLayout';

jest.mock('components/Rubric', () => 'Rubric');
jest.mock('./AssessmentContent', () => 'AssessmentContent');

describe('<AssessmentContentLayout />', () => {
  const props = {
    submission: 'submission'
  };

  it('render', () => {
    const wrapper = shallow(<AssessmentContentLayout {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('Rubric')).toHaveLength(1);
  });
});
