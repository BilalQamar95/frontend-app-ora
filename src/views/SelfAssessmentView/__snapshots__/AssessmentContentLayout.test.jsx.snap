// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssessmentContentLayout /> render 1`] = `
<div
  className="assessment-content-layout mr-auto ml-auto"
>
  <div
    className="content-wrapper"
  >
    <Row
      className="flex-nowrap m-0"
    >
      <Col
        className="p-0"
      >
        <AssessmentContent
          submission="submission"
        />
      </Col>
      <Rubric
        isGrading={true}
      />
    </Row>
  </div>
</div>
`;
